ui/src/main.ts

```diff
@@ -3,10 +3,30 @@ import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
 
 import {AppModule} from './app/app.module';
 import {environment} from './environments/environment';
+import { ConfigurationLoader as configurationLoaderPromise } from './environments/configurationLoader';
+
 
 if (environment.production) {
   enableProdMode();
 }
 
-platformBrowserDynamic().bootstrapModule(AppModule)
-  .catch(err => console.log(err));
+configurationLoaderPromise
+  .catch((err) => {
+    //TODO: Stop the app from moving forward here and show error about retrieving the file (snackBar(err))
+  })
+  .then((envfile) => {
+    return JSON.parse(envfile);
+  })
+  .catch((SyntaxError) => {
+    console.log(SyntaxError); // TODO: should stop the app here and show error about parsing the content snackBar(SyntaxError)
+  })
+  .then((env) => {
+    Object.entries(env).forEach(([key, value]) => {
```

Comment:
I would probably put this logic one level down in ConfigurationLoader. I don't think it should be returning  raw response text.